admin_manual/windmill_workflows/index.rst

```diff
@@ -0,0 +1,71 @@
+
+Nextcloud Scripts
+-----------------
+
+Nextcloud makes available a variety of scripts to be used in Windmill for interfacing with Nextcloud apps. You can find them
+at https://hub.windmill.dev/integrations/nextcloud and https://hub.windmill.dev/integrations/nextcloud/approvals or in your windmill instance when selecting existing scripts for creating a new workflow.
+
+Authentication
+~~~~~~~~~~~~~~
+
+All bricks have the option to use "AppAPI Authentication" or normal authentication using a Nextcloud resource in Windmill. When using normal authentication you will need to provide the correct password or app password of the user on behalf of whom you want to execute the script. When using "AppAPI Authentication" you can impersonate any Nextcloud user. This will only work when using the ExApp-packaged version of windmill.
+
+Passing values between blocks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When specifying script inputs you can either fill the parameters with static values or make references to the workflow input and the previous workflow steps.
+
+In order to reference the workflow input, use the ``flow_input`` variable. For example, ``flow_input.event.form.hash`` will reference the hash of a form from a nextcloud Forms event.
+
+Each step in a workflow is automatically assigned a letter identifier. In order to reference results from previous steps in your parameters, use the ``results`` variable with the id of the step to reference as a sub property. For example, use ``results.e.submission.answers`` to use the answers of of a form submission retrieved via the "Get form submission from Nextcloud Forms" script identified with the letter "e".
+
+Approval/Suspend steps
+~~~~~~~~~~~~~~~~~~~~~~
+
+Windmill allows using so-called approval steps, which are essentially asynchronous scripts that wait for the call to an additional webhook URL. The most prominent use case for this are approval workflows where you get automated input from somewhere which needs to be approved by a human. Once the human approves or disapproves by triggering the webhook URL the workflow will resume.
+
+Using the scripts provided for Nextcloud, you can send approval links to the humans in charge of approving via Nextcloud Talk or a simple notification in Nextcloud. Of course, you may also use any of the other scripts for sending messages available in the Windmill hub.
+
+Windmill has a default approval user interface at a specific URL, but it is not very nice. We recommend using the ``approve_links`` apps which allows creating a beautiful temporary approval page with a custom message and approve and disapprove buttons.
```

Comment:
Let's mention the app page: https://apps.nextcloud.com/apps/approve_links

Maybe it's also useful to explain that scripts of the "approval" kind still need a manual configuration when used in a workflow. In the workflow edition screen, select the approval script and in the bottom right pan, go in the "Advanced" tab, "Suspend" sub tab and check "Suspend/Approval/Prompt":

![approval script suspend flag](https://github.com/user-attachments/assets/0152fd14-22a1-400d-82d8-ed513b5fcd1f)